### DISCUSSION TITLE
Will version 6 keep the `request` where `Parameters` is just `[String: Any]`?

### DISCUSSION THREAD

**Discussion by funcJeff** [Alamofire/Alamofire#3696] [Q&A] [answered]
I know the fancy new thing is `request<Parameters: Encodable>` but I am hoping that the old `request` in which `Parameters` is  just a typealias for a dictionary will remain.

I know in version 5 that the `request` signature I am asking about is **not** deprecated, and so  it will probably still be there in 6, but will it be deprecated in 6?

I am hoping that it is **not** deprecated because the way we are marshaling parameters in our app would have to be reworked extensively to create types that conform to `Encodable`, and I hate using deprecated stuff. So I would like to see if anyone knows before 6 releases.

Thanks.

**Comment by jshier** [comment, marked as answer]
It will be yes, as maintaining two encoding paths for parameters is untenable. However, that won't happen until I can create an easier migration path for people, perhaps through the integration of an `AnyEncodable` type or something like that. There may be other solutions where I can manually encode `[String: Any]` parameters, but full `Codable` support is the future.

Additionally, Alamofire 6 is a ways out, as I haven't finished all of the 5 features I want yet.

**Reply by funcJeff** [reply]
> It will be yes

I can tell what you meant by context, but to be certain, I am confirming that you are saying that the `request` func which takes a `[String : Any]` for `parameters` **will** be deprecated in version 6.

**Reply by jshier** [reply]
Technically it will likely be deprecated in some transitionary versions of 5 and removed in 6, but I'll try to have a path forward for existing users. Not something you need to worry about for more than a year, most likely.
